### PR TITLE
axp - Chapter 8 - Filehandle References

### PR DESCRIPTION
**Pull request by a-panayotopoulos**
All fairly straightforward.

### CODE REVIEW

File: chapter08/axp/ex01 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+my ( $str, %opts );
+
+getopts( "sf:", \%opts );
+
+die "No filename given after -f\n" if exists $opts{f} && !defined $opts{f};
+die "Neither -f nor -s option passed\n" if !%opts;
+
+my $scalar_handle = $opts{s} ? IO::Scalar->new( \$str ) : IO::Null->new;
+my $file_handle = $opts{f} ? IO::File->new( $opts{f}, 'w' ) : IO::Null->new or die "Couldn't write to file: $!\n";
```
Comment (alexdeas) [inline chapter08/axp/ex01]: Neat use of IO::Null. Does this ``or`` split the line in the way you want though?

File: chapter08/axp/ex03 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env perl
+
+use 5.014;
+use warnings;
+
+foreach ( map { glob } @ARGV ) {
+	opendir my $dh, $_ or say( "Not a directory: $_" ) && next;
```
Comment (alexdeas) [inline chapter08/axp/ex03]: The exercise seems to want a function that uses this directory handle.

File: chapter08/axp/ex01 (hunk 1 above)
Comment (a-panayotopoulos) [inline chapter08/axp/ex01]: Yep. `or` has lower associativity than any other operator, including the ternary.
http://perldoc.perl.org/perlop.html#Operator-Precedence-and-Associativity

File: chapter08/axp/ex03 (hunk 2 above)
Comment (a-panayotopoulos) [inline chapter08/axp/ex03]: Yep, I've used [`readdir`](http://perldoc.perl.org/functions/readdir.html).

**Pull request merged**